docs/learn/first-dapp.mdx

```diff
@@ -0,0 +1,767 @@
+
+1. Finally, we will handle the deposit and withdraw transactions! In this last part, we will need to prepare and send a transaction. We will use the `useWriteContract` hook.
+
+```tsx
+import { useAccount, useReadContracts, useWriteContract } from "wagmi";
+import { Check, ChevronsUpDown, Loader2 } from "lucide-react";
+
+// ...
+
+// StakingCard:
+const { writeContract, isPending } = useWriteContract();
+
+return (
+	//...
+	
+  <Button
+    className="w-full mt-4"
+    onClick={() => {}}
+    disabled={disabled || amountStakable === 0}
+  >
+    Stake
+  </Button>
+{(data?.[3] ?? 0) < amountToStake * 10 ** (data?.[0] ?? 18) ? (
+  <Button
+    className="w-full mt-4"
+    onClick={() =>
+      writeContract({
+        abi: erc20Abi,
+        address: token,
+        functionName: "approve",
+        args: [
+          CONTRACT_ADDRESS,
+          BigInt(amountToStake * 10 ** (data?.[0] ?? 18)),
+        ],
+      })
+    }
+    disabled={disabled || amountStakable === 0 || isPending}
+  >
+    {isPending ? (
+      <Loader2 className="animate-spin" />
+    ) : (
+      <p>Approve</p>
+    )}
+  </Button>
+) : (
+  <Button
+    className="w-full mt-4"
+    onClick={() =>
+      writeContract({
+        abi: CONTRACT_ABI,
+        address: CONTRACT_ADDRESS,
+        functionName: "stake",
+        args: [token, amountToStake * 10 ** (data?.[0] ?? 18)],
+      })
+    }
+    disabled={disabled || amountStakable === 0 || isPending}
+  >
+    {isPending ? (
+      <Loader2 className="animate-spin" />
+    ) : (
+      <p>Stake</p>
+    )}
+  </Button>
+)}
+  
+	//...
+	
+  <Button
+    className="w-full mt-4"
+    onClick={() => {}}
+    disabled={disabled || amountStaked === 0}
+  >
+    Unstake
+  </Button>
+  <Button
+    className="w-full mt-4"
+    onClick={() =>
+      writeContract({
+        abi: CONTRACT_ABI,
+        address: CONTRACT_ADDRESS,
+        functionName: "withdraw",
+        args: [token, amountToUnstake * 10 ** (data?.[0] ?? 18)],
+      })
+    }
+    disabled={disabled || amountStaked === 0 || isPending}
+  >
+    {isPending ? <Loader2 className="animate-spin" /> : <p>Unstake</p>}
+  </Button>
+  
+	//...
+)
+```
+
+And voilà! 🎉 You should be able to easily stake and unstake from the list of allowlisted tokens.
+
+This frontend part only demonstrated read and write operations. Most Wagmi hooks return lots of data such as `isLoading` or `error`, which are useful for building reactive and beautiful frontends, so be sure to check the documentation. We also omit error handling here for simplicity.
+
+Lastly, we can listen for events to dynamically update the frontend. I'll leave this to you as an optional exercise if you want to go further (hint: you can use [`watchContractEvent`](https://wagmi.sh/core/api/actions/watchContractEvent) or [watchAsset](https://wagmi.sh/core/api/actions/watchAsset))
+
+---
```

Comment:
this doesn't appear to render, so might be good to switch out with some other kind of divider, or just delete altogether. 

I think it's being interpreted as the opening of a YAML front matter block (i.e. as in line 1).